File: backend/src/SpecialPrograms.Api/Models/Student.cs
namespace SpecialPrograms.Api.Models;

public class Student
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public string LocalId { get; set; } = string.Empty;
    public string FirstName { get; set; } = string.Empty;
    public string LastName { get; set; } = string.Empty;
    public DateTime DateOfBirth { get; set; }
    public string GradeLevel { get; set; } = string.Empty;
    public string Campus { get; set; } = string.Empty;
    public string GuardianContact { get; set; } = string.Empty;
    public string ProgramFocus { get; set; } = string.Empty;
    public DateTime EnrollmentDate { get; set; } = DateTime.UtcNow.Date;
    public DateTime? NextReviewDate { get; set; }
    public ICollection<NeedsAssessment> NeedsAssessments { get; set; } = new List<NeedsAssessment>();
    public ICollection<Goal> Goals { get; set; } = new List<Goal>();
}

File: frontend/teks-mvp/src/app/models/student.ts
export interface StudentSummary {
  id: string;
  localId: string;
  firstName: string;
  lastName: string;
  gradeLevel: string;
  campus: string;
  programFocus: string;
  enrollmentDate: string;
  nextReviewDate?: string;
  needsAssessments: number;
  goals: number;
  activeGoals: number;
}

export interface StudentDetail {
  id: string;
  localId: string;
  firstName: string;
  lastName: string;
  gradeLevel: string;
  campus: string;
  programFocus: string;
  enrollmentDate: string;
  nextReviewDate?: string;
  dateOfBirth: string;
  guardianContact: string;
  needsAssessments: NeedsAssessment[];
  goals: GoalDetail[];
}

export interface NeedsAssessment {
  id: string;
  studentId: string;
  academicNeeds: string;
  supportServices: string;
  instructionalStrategies: string;
  assessmentTools: string;
  createdAt: string;
}

export interface Goal {
  id: string;
  studentId: string;
  description: string;
  category: string;
  measurement: string;
  owner: string;
  targetDate: string;
  status: string;
}

export interface GoalDetail extends Goal {
  progressUpdates: ProgressUpdate[];
}

export interface ProgressUpdate {
  id: string;
  goalId: string;
  summary: string;
  outcome: string;
  evidenceUrl: string;
  nextAction: string;
  recordedBy: string;
  recordedAt: string;
}
